numalogic/models/autoencoder/variants/conv.py

```diff
@@ -20,84 +20,271 @@
 from numalogic.models.autoencoder.base import BaseAE
 
 LOGGER = logging.getLogger(__name__)
+T = TypeVar("T", bound="Tensor")
```

Comment:
I think we can move this to our tools/types.py

Reply:
I removed it for now, let's just go with Tensor